mexican_generator.py

```diff
@@ -0,0 +1,259 @@
+import json
+import queue as python_queue
+import copy
+from enum import IntEnum
+from pprint import pprint
+import collections.abc
+from typing import Union
+
+DEBUG = False
+
+
+def write_cgs(file, cgs):
+    def set_default(obj):
+        if isinstance(obj, set):
+            return list(obj)
+        raise TypeError
+
+    with open(file, 'w+') as o:
+        o.writelines(json.dumps(cgs, default=set_default))
+
+
+class CGS:
+    def __init__(self, player_count=0, labeling=[], transitions={}, moves=[]):
+        self.game_struct = {"player_count": player_count,
+                            "labeling": labeling,
+                            "transitions": transitions,
+                            "moves": moves}
+
+    def add_trans(self, q, move, result_state):
+        # shifting up by once so moves are in range {0, 1, 2} and mapping 'other' to 'right' (convert to list for edit)
+        move = list(move)
+
+        for i, m in enumerate(move):
+            if m == 2:
+                move[i] = MexicanMoves.RIGHT
+
+        entry = {move[0] + 1: {move[1] + 1: {move[2] + 1: {result_state}}}}
+        try:
+            self.game_struct['transitions'][q]
+        except KeyError:  # maybe q doesn't exist
+            self.game_struct['transitions'].update({q: {}})
+        finally:
+            self.game_struct['transitions'][q] = update(self.game_struct['transitions'][q], entry)
+
+    def append_label(self, labels):
+        self.game_struct['labeling'].append(labels)
+
+    def append_move(self, moves):
+        self.game_struct['moves'].append(moves)
+
+
+class State:
+    def __init__(self, state_int):
+        # Unroll base 10 rep to binary rep with leading zeroes
+        bin_rep = list(map(int, get_binary_rep(state_int)))
+        if len(bin_rep) == 1:
+            bin_rep = [0] + bin_rep
+
+        if len(bin_rep) == 2:
+            bin_rep = [0] + bin_rep
+
+        self.state = bin_rep
+
+    def __eq__(self, other):
+        return self.state == other.state
+
+    def __str__(self):
+        return str(self.state)
+
+    def base10_rep(self):
+        return get_base10_rep(self.__str__())
+
+
+class MexicanMoves(IntEnum):
+    WAIT = 0
+    LEFT = -1
+    RIGHT = 1
+    OTHER = 2
```

Comment:
Why isn't this just 0..3?

Reply:
Because of the kill-logic, etc. It's mapped to [0..n) before writing.